File: Minimum_Subsequence_in_Non_Increasing_Order.py
nums = [4,3,10,9,8]
# # nums = [4,4,7,6,7]


def minSubsequence(nums):
    nums.sort()
    sumOfVal = sum(nums)
    ls= []
    for i in range(len(nums)-1,-1,-1):
        ls.append(nums[i])
        if sum(ls) > sumOfVal-sum(ls):
            return ls
    return ls
    
   
print(minSubsequence(nums))

File: Minimum_Subsequence_in_Non_Increasing_Order.go
package main

import (
	"fmt"
	"sort"
)

func main() {
	nums := []int{4, 3, 10, 9, 8}
	fmt.Println(minSubsequence(nums))

}

func minSubsequence(nums []int) []int {
	sort.Ints(nums)
	ans := []int{}
	totalSum := 0
	curVal := 0
	for _, val := range nums {
		totalSum += val
	}
	for i := len(nums) - 1; i > -1; i-- {
		curVal += nums[i]
		ans = append(ans, nums[i])

		if curVal > totalSum-curVal {
			break
		}
	}
	return ans
}
